src/render.c

```diff
@@ -28,6 +28,8 @@ int render_deco_height(void) {
     int deco_height = config.font.height + 4;
     if (config.font.height & 0x01)
         ++deco_height;
+    if (config.hide_combined_titles)
```

Comment:
This is not the correct place to decide this. This option should only affect tabbed / stacked containers *and* only if their border is set to anything other than `normal` (i.e., either `pixel` or `none`). So I think the logic has to go into `con_render_tabbed` and similar.

Also, I think `con_border_style` has to be adapted as well. Once all of that is done please test this with various border settings and window layouts (nested tabbed containers, floating etc.) and different configs (enabled window borders etc.).